plugins/janus_audiobridge.c

```diff
@@ -6819,6 +6916,99 @@ static void *janus_audiobridge_handler(void *data) {
 	JANUS_LOG(LOG_VERB, "Leaving AudioBridge handler thread\n");
 	return NULL;
 }
+static void janus_audiobridge_rec_add_wav_header(janus_audiobridge_room *audiobridge) {
+
+	/* Do we need to record the mix? */
+	char filename[255];
+	gint64 now = janus_get_real_time();
+	audiobridge->rec_start_time = now;
+	if(audiobridge->record_file) {
+		g_snprintf(filename, 255, "%s-%"SCNi64"%s%s", audiobridge->record_file,
+			now, rec_tempext ? "." : "", rec_tempext ? rec_tempext : "");
+	} else {
+		g_snprintf(filename, 255, "janus-audioroom-%s-%"SCNi64".wav%s%s", audiobridge->room_id_str,
+			now, rec_tempext ? "." : "", rec_tempext ? rec_tempext : "");
+	}
+	audiobridge->recording = fopen(filename, "wb");
+	if(audiobridge->recording == NULL) {
+		JANUS_LOG(LOG_WARN, "Recording requested, but could NOT open file %s for writing...\n", filename);
+	} else {
+		JANUS_LOG(LOG_VERB, "Recording requested, opened file %s for writing\n", filename);
+		/* Write WAV header */
+		wav_header header = {
+			{'R', 'I', 'F', 'F'},
+			0,
+			{'W', 'A', 'V', 'E'},
+			{'f', 'm', 't', ' '},
+			16,
+			1,
+			audiobridge->spatial_audio ? 2 : 1,
+			audiobridge->sampling_rate,
+			audiobridge->sampling_rate * 2,
+			2,
+			16,
+			{'d', 'a', 't', 'a'},
+			0
+		};
+		if(fwrite(&header, 1, sizeof(header), audiobridge->recording) != sizeof(header)) {
+			JANUS_LOG(LOG_ERR, "Error writing WAV header...\n");
+		}
+		fflush(audiobridge->recording);
+		audiobridge->record_lastupdate = janus_get_monotonic_time();
+		g_atomic_int_set(&audiobridge->wav_header_added, 1);
+	}
+}
+static void janus_audiobridge_update_wav_header(janus_audiobridge_room *audiobridge) {
+
+	/* Update the length in the header */
+	fseek(audiobridge->recording, 0, SEEK_END);
+	long int size = ftell(audiobridge->recording);
+	if(size >= 8) {
+		size -= 8;
+		fseek(audiobridge->recording, 4, SEEK_SET);
+		fwrite(&size, sizeof(uint32_t), 1, audiobridge->recording);
+		size += 8;
+		fseek(audiobridge->recording, 40, SEEK_SET);
+		fwrite(&size, sizeof(uint32_t), 1, audiobridge->recording);
+		fflush(audiobridge->recording);
+	}
+	fclose(audiobridge->recording);
+	audiobridge->recording = NULL;
+	g_atomic_int_set(&audiobridge->wav_header_added, 0);
+
+	char filename[255];
+	if(audiobridge->record_file) {
+		g_snprintf(filename, 255, "%s-%"SCNi64, audiobridge->record_file,audiobridge->rec_start_time);
```

Comment:
This should just be `"%s"`, not `"%s-%"SCNi64`: `record_file` contains the `.wav` extension already, which means you'd be appending the time _after_ the extension, breaking it. There's no need to add the current time to the filename when it's provided by the end user: as explained in another comment, in that case it's the user's responsibility to provide unique filenames, or risk overwriting previous recordings otherwise.

Reply:
I wanted to record the timestamp in filename so that it can be used for audio video synchronisation, i fear if timestamp is added in with the API call, the delay between API call may not reflect the exact time.
Can we add ".wav" after the timestamp ?

Reply:
Again, that's not something YOU have to worry about. If people provide `record_file`, than they may put the timestamp in the filename yourself. If you don't provide it, then your pattern does the job.

>  Can we add ".wav" after the timestamp ?

No, it would be a breaking change (since yout changes impact how recordings are started after `create` too).